File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:

vector<int> findDisappearedNumbers(vector<int>& nums) {
    sort(nums.begin(),nums.end());
      int size = nums.size(); vector<int>sam;
    for(int i=1;i<=size;i++){
      if(!(binary_search(nums.begin(),nums.end(),i))){
            sam.push_back(i); }}            
      return sam;       
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        # i am using cyclic sort
        i , ans = 0 , []
        while i < len(nums):
            index = nums[i] -1
            if (nums[i] != nums[index]):
                # swap
                nums[i] , nums[index]  = nums[index] , nums[i]
            else:
                i += 1
        for i , item in enumerate(nums):
            if i+1 != item:
                ans.append(i+1)
        return ans
